### ISSUE TITLE
Sources with 0 dec

### ISSUE THREAD

**Issue by AlecThomson** [CIRADA-Tools/RMTable#3]
About 2% of sources seem to have _exactly_ 0deg declination in the master catalogue v0.1.7. The majority seem to be from the Taylor catalogue, but there are a few other in there as well. Converting from the provided Galactic coordinates gives small, but >0 declination values.

I'm guessing it's a safe workaround to use the Galactic coordinates for now, but it'd be good if we could isolate the source of this error.

**Comment by zhangxiang-planet** [comment]
Follow up this issue: we did a crossmatch between the equatorial coordinates and galactic coordinates. At a separation limit of 1 arcsec, there are about 7% sources without matches. The distribution of these sources is given in the equatorial frame:

<img width="548" alt="Screen Shot 2020-09-10 at 4 22 31 pm" src="https://user-images.githubusercontent.com/22098261/92702478-d2231000-f383-11ea-837d-adb3adeeca1e.png">

**Comment by Cameron-Van-Eck** [comment]
Thanks a lot for finding that! I'll track that down today (+tomorrow if needed).

**Comment by Cameron-Van-Eck** [comment]
Hey @AlecThomson , @zhangxiang-planet !

I have tracked this down, I think. The case of the sources with zero declination was traced back to problems in reading the original machine-readable tables (turns out not all machine-readable tables are equally machine-readable - who knew?). Also, tracking down the reasons why galactic and equatorial coordinates didn't match up was an interesting rabbit hole. Turned out there were at least 4 reasons for these problems:
1) sources with 00 degrees declination were failing my hacked-together sexagesimal conversion, causing them to be put at exactly 0 dec.
2) sources with `-00` degrees declination were losing their sign on read-in, causing them to still fail even after I fixed problem 1.
3) Due to the rather hack-y way I read in certain tables, catalogs that stored declination with integers for the number of arcseconds converted the declinations into integer degrees, producing wrong declinations.

All 3 of these are now fixed, as near as I can tell. The remaining populations of sources that are slightly offset between Galactic and equatorial are because:
4) Some catalogs that supplied both galactic and equatorial used different software to make the conversion (and/or used too few significant digits in one or both frames), resulting in differences greater than 1 arcsec.
For now I have decided not to change these, in the spirit of being true to the original data (and also not necessarily knowing which set of coordinates are 'true' and which are off slightly). So these will remain a problem, meaning that ultimately we'll never be able to use cross-matching thresholds smaller than a few arcsec.

This problem has reminded me of why I hate sexagesimal with a passion. I once again feel justified in rejecting the use of sexagesimal coordinates in the catalog, which was requested.

Once again, thanks a lot for finding these problems.
Cheers,
Cameron
